File: Implementation/App/Server/Models/Stack.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AspCoreServer.Models
{

    public class Stack
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int CellCount { get; set; }
        public CircuitType CircuitType { get; set; }

        public ICollection<BatteryStack> BatteryStacks { get; set; }
        public ICollection<StackCell> StackCells { get; set; }
        public ICollection<Measurement> Measurements { get; set; }

        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [DataType(DataType.Date)]
        public DateTime Created { get; set; }

        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [DataType(DataType.Date)]
        public DateTime Modified { get; set; }

        //Setting Default value
        public Stack()
        {
            Description = "Beschreibung";
            Created = DateTime.Now;
            Modified = DateTime.Now;
        }
    }

}

File: App/Client/app/models/Stack.ts
import { BatteryStack } from './BatteryStack';
import { StackCell } from './StackCell';
import { CircuitType } from './Battery';
import { Measurement } from './Measurement';

export interface Stack {
    id: number;
    name: string;
    description: string;
    cellCount: number;
    circuitType: CircuitType;

    batteryStacks: BatteryStack[];
    stackCells: StackCell[];
    measurements: Measurement[];

    created: Date;
    modified: Date;

    active: Boolean;
}
